### PR TITLE
tooling update: vitest & jest types

### PR DESCRIPTION
**Pull request by snarkipus**
Had a testing issue that necessitated updating Vitest. This change updates to the latest vitest (0.27.x) and updates the jest types.

> **deps.interopDefault**
> Interpret CJS module's default as named exports.

[Vitest Docs](https://vitest.dev/config/#deps-interopdefault)

### CODE REVIEW

**Review by kaisalmen — APPROVED** [review]
Don't mind my comment. We have to fix the globals problem directly by adjusting imports specifically in the test. I have opened #873 for this. I guess we stay with vitest and therefore it is a good idea.

File: package.json
Comment (snarkipus) [reply package.json:23]: @kaisalmen I'll defer to @msujew for the [specifics](https://github.com/langium/langium/pull/690#pullrequestreview-1115186651) from PR #690 , but switching to `vitest/globals` results in a linting supernova.

Philosophically, I'm _ok_ with discrete imports on a per-file basis since that's more explicit, but that's a decision I'll defer to the maintainers. I toyed with the idea originally, but the Langium team implemented some fairly elegant test harnesses that I didn't fully understand at the time so I gave up. Theoretically, the jest-style `vitest/globals` would preserve the current test workflow but resolving the typing/linting mess is beyond my skill.

**Review by msujew — APPROVED** [review]
The changes also look good to me 👍

**Pull request merged**
